tools/topology/sof-cml-eq-rt5682-kwd.m4

```diff
@@ -0,0 +1,233 @@
+#
+# Topology for Cometlake with rt5682 codec and Keyword Detect.
+#
+
+# Include topology builder
+include(`utils.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+include(`ssp.m4')
+
+# Include TLV library
+include(`common/tlv.m4')
+
+# Include Token library
+include(`sof/tokens.m4')
+
+# Include Cannonlake or Icelake DSP configuration
+ifelse(PLATFORM, `icl', include(`platform/intel/icl.m4'),
+	ifelse(PLATFORM, `whl', include(`platform/intel/cnl.m4'),
+		ifelse(PLATFORM, `cml', include(`platform/intel/cnl.m4'), `')))
+include(`platform/intel/dmic.m4')
+
+define(KWD_PIPE_SCH_DEADLINE_US, 20000)
+
+DEBUG_START
+
+undefine(`SSP_INDEX')
+define(`SSP_INDEX', ifelse(PLATFORM, `icl', `0',
+	ifelse(PLATFORM, `whl', `1',
+		ifelse(PLATFORM, `cml', `0', `'))))
+
+undefine(`SSP_NAME')
+define(`SSP_NAME', ifelse(PLATFORM, `icl', `SSP0-Codec',
+	ifelse(PLATFORM, `whl', `SSP1-Codec',
+		ifelse(PLATFORM, `cml', `SSP0-Codec', `'))))
+
+undefine(`SSP_MCLK_RATE')
+define(`SSP_MCLK_RATE', ifelse(PLATFORM, `icl', `19200000',
+	ifelse(PLATFORM, `whl', `24000000',
+		ifelse(PLATFORM, `cml', `24000000', `'))))
+
```

Comment:
I'd prefer to see this taken care of by ```include(PLATFORM.m4)``` where PLATFORM.m4 sets the correct SSP port, name and clock

Reply:
@fredoh9 this would be good if you could make the same change to all topologies that have machine specific conditional logic as it would be a great simplification as a separate PR

Reply:
@lgirdwood Sure, I will review PLATFORM specific changes.